swap_meet/vendor.py

```diff
@@ -1,2 +1,167 @@
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        """Swaps items between the current vendor and another vendor.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap items with.
+            my_item (Item): The item the current vendor wants to trade.
+            their_item (Item): The item the other vendor wants to trade.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+
+        """
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.remove(my_item)
+        other_vendor.remove(their_item)
+
+        other_vendor.add(my_item)
+        self.add(their_item)
+
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        """Swaps the first item of the current vendor's inventory with the first item of the other vendor's inventory.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap with.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+        """
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_item = self.inventory[0]
+        their_item = other_vendor.inventory[0]
+
+        return self.swap_items(other_vendor, my_item, their_item)
+
+    def get_by_category(self, category):
+        """Returns a list of items in the inventory that belong to a specific category.
+
+        Parameters:
+            category (str): The category to filter items by.
+
+        Returns:
+            list: A list of items that match the given category.
+        """
+        # item_list = []
+        # for item in self.inventory:
+        #     if item.get_category() == category:
+        #         item_list.append(item)
+        # use list comprehension to simplify code
+        return [item for item in self.inventory if item.get_category() == category]
+    
+    def get_best_by_category(self, category):
+        """Returns the item with the best condition in a given      category.
+
+        Parameters:
+            category (str): The category to search for the best item.
+
+        Returns:
+            Item or None: The item with the best condition in the specified category, or None if no items match.
+        """
+        # get items with matching category
+        item_list = self.get_by_category(category)
+        if not item_list:
+            return None
+        best_condition = item_list[0].condition
+        best_item = item_list[0]
```

Comment:
This is a small thing, but we are technically duplicating data by holding onto the `best_condition` since that value is already an attribute within `best_item`. In situations like this, it can often be a good idea to just grab the item and then make comparisons against that item's condition. So instead of `best_condition` we would just use `best_item.condition`.